nemo/collections/nlp/models/nlp_model.py

```diff
@@ -95,6 +95,10 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None, no_lm_init=False):
                 and cfg.get('tokenizer').get('vocab_file')
             ):
                 vocab_file = self.register_artifact('tokenizer.vocab_file', cfg.tokenizer.vocab_file)
+        
+        if (not AppState().is_model_being_restored and AppState().restore_attr_is_model_being_restored ):
+            AppState().is_model_being_restored = True
```

Comment:
I don't understand why you have to create the attribute `restore_attr_is_model_being_restored` . Is the issue that AppState().is_model_being_restored not set properly when restoringthe model?

Reply:
yes , before restoring we explicitly set the is_model_being_restored to true using function set_model_restore_state . But while restoring it gets overwritten to false (inside if block at line 70 - if cfg.get('tokenizer'):).
That is why this new attribute is created restore_attr_is_model_being_restored which is by default false but if we set it to true then it will restore true state of is_model_being_restored.

Reply:
why do you want to manually modify `is_model_being_restored`? It seems that this variable is temporally used when loading the checkpoint file. 
```
    @classmethod
    def load_from_checkpoint(
        cls,
        checkpoint_path: str,
        *args,
        map_location: Optional[Union[Dict[str, str], str, torch.device, int, Callable]] = None,
        hparams_file: Optional[str] = None,
        strict: bool = True,
        **kwargs,
    ):
        """
        Loads ModelPT from checkpoint, with some maintenance of restoration.
        For documentation, please refer to LightningModule.load_from_checkpoint() documentation.
        """
        checkpoint = None
        try:
            cls._set_model_restore_state(is_being_restored=True)

            checkpoint = super().load_from_checkpoint(
                checkpoint_path=checkpoint_path,
                *args,
                map_location=map_location,
                hparams_file=hparams_file,
                strict=strict,
                **kwargs,
            )

        finally:
            cls._set_model_restore_state(is_being_restored=False)
        return checkpoint

```